### ISSUE TITLE
[dynamo] Missing guard for builtin callable

### ISSUE THREAD

**Issue by anijain2305** [pytorch/pytorch#118156]
### 🐛 Describe the bug

Following test fails because Dynamo fails to recognize that `sum` builtin has been patched

~~~python
import torch

def fn(x):
    return sum(x)


opt_fn = torch.compile(fn, backend="eager")
x = torch.randn(4)
assert torch.allclose(fn(x), opt_fn(x))

# Patch sum
sum = abs
assert torch.allclose(fn(x), opt_fn(x))
~~~

### Versions

N/A

cc @ezyang @msaroufim @bdhirsh @zou3519 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

**Comment by mdevino** [comment]
Hello,
if this is not already taken, could I get some guidance to start working on this issue?

I understand the issue but I'm not quite sure where to start. My understanding is that I should start looking into torch/_dynamo/guards.py. Is this correct?

This would be my first contribution.

**Comment by mdevino** [comment]
Hi @laithsakka, @anijain2305,
sorry to ping you. I just wanted to double-check if it would be okay for me to work on this issue.

In the meantime, I'm watching [this talk](https://www.youtube.com/live/5FNHwPIyHr8?si=_ctK1-kmqDIvLggZ) and reading PyTorch's documentation to better understand PyTorchDynamo.

**Comment by laithsakka** [comment]
@mdevino yes you can take the issue but make sure that its still valid, I recall that @anijain2305  told me he want to make sure first its an issue

**Comment by anijain2305** [comment]
Yes, this is more of a performance question. We intentionally skip guards on global functions today. 
I removed that skip in https://github.com/pytorch/pytorch/pull/119847 and started a perf run to see the impact.

@mdevino Maybe this is not the right task to start. There are other tasks with label "dynamo-must-fix" or "good-first-issue" if you are interested.

**Comment by mdevino** [comment]
Hello,

sure. Thanks for letting me know!
I'll look into other issues to start contributing.

**Comment by anijain2305** [comment]
This is fixed now.
